Tools/Scripts/webkitpy/benchmark_runner/http_server_driver/http_server/builtin_http_server.py

```diff
@@ -62,10 +63,13 @@ def do_POST(self):
     parser.add_argument('--interface', default='')
     parser.add_argument('--log-path', default='/tmp/run-benchmark-http.log')
     args = parser.parse_args()
+    addr_info = socket.getaddrinfo(args.interface, args.port)
+    addr_pair = addr_info[0][4]
```

Comment:
I think we can use the first address returned by `socket.getaddrinfo` because from what `simple_http_server_driver` suggests, the interface should be passed in a IP address over a domain which should only return 1 option. Additionally this is the similar method which Twisted uses when handling an IPV6 address: https://github.com/twisted/twisted/blob/trunk/src/twisted/internet/tcp.py#L649

If there is preference for IPV6 addresses then I think that could be saved for a future feature if domains are being passed in.

I agree that we should raise our own error messages on socket.gaierror exceptions.